extensions/import/src/test/wizard/pages/modifyColumnsPage.test.ts

```diff
@@ -0,0 +1,111 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as TypeMoq from 'typemoq';
+import * as azdata from 'azdata';
+import { ApiWrapper } from '../../../common/apiWrapper';
+import * as constants from '../../../common/constants';
+import { FlatFileWizard } from '../../../wizard/flatFileWizard';
+import * as should from 'should';
+import { ModifyColumnsPage } from '../../../wizard/pages/modifyColumnsPage';
+import { ImportDataModel } from '../../../wizard/api/models';
+import { TestImportDataModel, TestFlatFileProvider } from '../../utils.test';
+import { ImportPage } from '../../../wizard/api/importPage';
+import { FlatFileProvider } from '../../../services/contracts';
+
+describe('import extension modify Column Page', function () {
+	let wizard: azdata.window.Wizard;
+	let page: azdata.window.WizardPage;
+	let modifyColumnsPage: ModifyColumnsPage;
+	let mockFlatFileWizard: TypeMoq.IMock<FlatFileWizard>;
+	let mockImportModel: TypeMoq.IMock<ImportDataModel>;
+	let mockApiWrapper: TypeMoq.IMock<ApiWrapper>;
+	let pages: Map<number, ImportPage> = new Map<number, ImportPage>();
+	let mockFlatFileProvider: TypeMoq.IMock<FlatFileProvider>;
+
+	beforeEach(function () {
+		// Keeping the original behaviour of apiWrapper until some setup is needed to mock stuff
+		mockApiWrapper = TypeMoq.Mock.ofType(ApiWrapper, TypeMoq.MockBehavior.Loose);
+		mockApiWrapper.callBase = true;
+
+		mockFlatFileProvider = TypeMoq.Mock.ofType(TestFlatFileProvider);
+		mockFlatFileWizard = TypeMoq.Mock.ofType(FlatFileWizard, TypeMoq.MockBehavior.Loose, undefined, mockFlatFileProvider.object, mockApiWrapper.object);
+		mockImportModel = TypeMoq.Mock.ofType(TestImportDataModel, TypeMoq.MockBehavior.Loose);
+
+		wizard = mockApiWrapper.object.createWizard(constants.wizardNameText);
+		page = mockApiWrapper.object.createWizardPage(constants.page3NameText);
+
+	});
+
+	it('checking if all components are initialized properly', async function () {
+
+		await new Promise(function (resolve) {
+			page.registerContent(async (view) => {
+				modifyColumnsPage = new ModifyColumnsPage(mockFlatFileWizard.object, page, mockImportModel.object, view, TypeMoq.It.isAny(), mockApiWrapper.object);
+				pages.set(1, modifyColumnsPage);
+				await modifyColumnsPage.start().then(() => {
```

Comment:
The .then is doing nothing - just remove it. You can just call resolve directly after
```
await modifyColumnsPage.start();
resolve();
```